packages/darklang/prettyPrinter/programTypes.dark

```diff
@@ -458,21 +455,31 @@ module Darklang =
         | EInt(_id, i) -> Int.toString i
         | EFloat(_id, sign, whole, remainder) ->
           let signPart = PACKAGE.Darklang.PrettyPrinter.sign sign
-          let wholePart = Int.toString whole
-
+          let rem = $".{remainder}" |> Float.parse
           let remainderPart =
-            match remainder with
-            | Nothing -> ""
-            | Just r -> $".{Int.toString r}"
+              match rem with
+              | Ok r ->
+                  if r == 0.0 then ""
+                  else (r  |> Float.toString) |> String.dropFirst 2
+              | Error e -> e
+          let wholePart =
+              match whole with
+              | "0" -> if remainderPart != "" then "0" else ""
+              | _ -> whole
+          let formattedFloat =
```

Comment:
I'm finding it hard to understand what's happening here. Could you add comments (with examples maybe) to discuss the edge cases you're handling?

Also can you add tests? I think you should be able to add tests within backend/testfiles/execution.

Reply:
We don't need this anymore. it was used when the number lies between 0 and 1, to keep the "0" in whole part.